src/chain/chain-denominator.cc

```diff
@@ -231,6 +234,7 @@ void DenominatorComputation::Beta(int32 t) {
 }
 
 BaseFloat DenominatorComputation::Forward() {
+  NVTX_RANGE(__func__);
```

Comment:
is this debug code?

Reply:
No, this is instrumentation for the profiler. The `NVTX_RANGE` macro is set when compiled with `-DUSE_NVTX` otherwise it's empty.